File: WebAppSSR/src/app/core/http/dto/menu-dto/menu-analytics/menu-item-access.dto.ts
export interface MenuItemAccessDto {
    id: number;
    createdAt: string; // ISO date string (e.g., "2024-05-23T18:00:00Z")
    menuItemAccessType: 'favorite' | 'details'; // restrict to expected values
    menuId: number;
    menuCategoryId: number;
    menuItemId: number;
}

File: WebAPI/WebAPI.Application/DTO/MenuDTO/MenuAccess/MenuItemAccessDto.cs
namespace WebAPI.Application.DTO.MenuAccess;

public class MenuItemAccessDto
{
    public int Id { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public string MenuItemAccessType { get; set; } // "favorite" or "details"
    public int MenuId { get; set; }
    public int MenuCategoryId { get; set; }
    public int MenuItemId { get; set; }
}